pkglistgen/tool.py

```diff
@@ -620,67 +681,101 @@ def update_and_solve_target(
 
         group = target_config.get('pkglistgen-group', '000package-groups')
         product = target_config.get('pkglistgen-product', '000product')
+        productcompose = target_config.get('pkglistgen-product', '000productcompose')
         release = target_config.get('pkglistgen-release', '000release-packages')
         oldrepos = target_config.get('pkglistgen-repos', '000update-repos')
 
-        url = api.makeurl(['source', project])
-        packages = ET.parse(http_GET(url)).getroot()
-        if packages.find('entry[@name="{}"]'.format(product)) is None:
-            if not self.dry_run:
-                undelete_package(api.apiurl, project, product, 'revive')
-            # TODO disable build.
-            logging.info('{} undeleted, skip dvd until next cycle'.format(product))
-            return
-        elif not force:
-            root = ET.fromstringlist(show_results_meta(api.apiurl, project, product,
-                                                       repository=[main_repo], multibuild=True))
-            if len(root.xpath('result[@state="building"]')) or len(root.xpath('result[@state="dirty"]')):
-                logging.info('{}/{} build in progress'.format(project, product))
-                return
+        # Cache dir specific to hostname and project.
+        host = urlparse(api.apiurl).hostname
+        cache_dir = CacheManager.directory('pkglistgen', host, project)
 
-        drop_list = api.item_exists(project, oldrepos)
+        drop_list = []
         checkout_list = [group, product, release]
-        if drop_list and not only_release_packages:
-            checkout_list.append(oldrepos)
+        if not force:
+          root = ET.fromstringlist(show_results_meta(api.apiurl, project, product,
+                                                     repository=[main_repo], multibuild=True))
+          if len(root.xpath('result[@state="building"]')) or len(root.xpath('result[@state="dirty"]')):
+              logging.info('{}/{} build in progress'.format(project, product))
+              return
+        if git_url:
+            if os.path.exists(cache_dir + "/.git"):
+                # reset and update existing clone
+                logging.debug(subprocess.check_output(
+                    ['git', 'reset', '--hard'], cwd=cache_dir, encoding='utf-8'))
+                logging.debug(subprocess.check_output(
+                    ['git', 'pull', '--rebase'], cwd=cache_dir, encoding='utf-8'))
+            else:
+                if os.path.exists(cache_dir):
+                    shutil.rmtree(cache_dir)
+                logging.debug(subprocess.check_output(
+                    ['git', 'clone', '--recurse-submodules', git_url, cache_dir], encoding='utf-8'))
+            if os.path.exists(cache_dir + '/' + '000update-repos'):
```

Comment:
Technically the wrong way to check for this, but IMO ok for now, it's at least visible in the code